google/cloud/universe_domain/integration_tests/impersonation_tests.cc

```diff
@@ -105,6 +130,21 @@ TEST_F(DomainUniverseImpersonationTest, SAToSAImpersonationGrpc) {
   }
 }
 
+TEST_F(IdTokenServiceAccountImpersonationTest, SAToSAImpersonationRest) {
```

Comment:
personal style comment: I wouldn't use a test class if there is only 1 test. It just means I have to look in two places to see what the code is doing.

Also, I personally have several monkeys where I mixed up some local `foo` with the member `foo_`. That sort of thing is easy to do when we have member variables and impossible when we don't.

When the setup code is really simple, duplicating it across multiple tests is sometimes cleaner. In this case the additional set up is only two lines:

```
    auto key_file = gc::internal::GetEnv("UD_IDTOKEN_SA_KEY_FILE").value_or("");
    ASSERT_FALSE(key_file.empty());
```

But feel free to ignore this opinion, especially if you plan to add more of these tests.

Reply:
Yes, right. I will unify the test classes.

Reply:
Yes, right. I will unify the test classes.

Reply:
I meant doing work in the `TEST(...) { ... }` instead of the `SetUp()`, but no worries.